packages/k-ramel/doc/DRIVERS.md

```diff
@@ -16,3 +16,191 @@
     <a href="./STORE.md">store</a>
   &nbsp;]
 </p>
+
+## What is a driver?
+A driver is something that can access the whole store. We use it as something where we can write our side effects (accessing HistoryAPI for instance, or `fetch`, etc). But you can also use it to add some application wise logic, like service.
+
+Just keep in mind that drivers can't be splited or hot loaded. The more you add logic to drivers, the more your bundle will be bloated.
+If you want to add some logic code maybe you should use `listeners` in your application instead. So you can add and remove them on the fly.
+
+We use drivers to share our work too, you can find some in the [ecosystem documentation](../../../README.md#how-to-use-k-ramel).
+
+At the beginin, our drivers was inspired by [CycleJS description](https://cycle.js.org/drivers.html).
+
+## Example
+### Creating a driver
+**dumbDriver.js**
+```js
+export default {
+  getDriver: (store) => {
+    // your implementation
+
+    // your driver API
+    return {
+      // the store is the k-ramel store, it means you can use others drivers (!), custom dispatch,
+      // and our types wrapper
+      sayHi: () => {
+        // dispatch something
+        store.dispatch('@@dumbDriver/HI')
+
+        // modify state
+        store.data.add({ id: Date.now(), label: 'I said hi!' })
+      },
+    }
+  },
+}
+```
+
+### Attach your new driver to the store and give it a name
+**store.js**
+```js
+import { createStore, data } from 'k-ramel'
+import dumb from './dumbDriver'
+
+export default createStore(
+  // store definitions
+  {
+    data: types.keyValues(),
+  },
+  // store options
+  {
+    // drivers (key/values)
+    drivers: {
+      dumb,
+    },
+  },
+)
+```
+
+### Use a driver into a reaction
+**listeners.js**
+```js
+import { when } from 'k-ramel'
+
+// drivers API are accessible into the third argument of your reactions
+const myReaction = (action, store, drivers) => {
+  // get the dumb driver
+  const { dumb } = drivers
+
+  // use its API
+  dumb.sayHi()
+}
+
+export default [
+  when('@@krml/INIT')(myReaction),
+]
+```
+
+### Attach the reaction to store (via listeners)
+**store.js**
+```js
+import { createStore, data } from 'k-ramel'
+import dumb from './dumbDriver'
+import listeners from './listeners'
+
+export default createStore(
+  // store definitions
+  {
+    data: types.keyValues(),
+  },
+  // store options
+  {
+    // root listeners
+    listeners, //////// <------ modification is here 👋
+    // drivers (key/values)
+    drivers: {
+      dumb,
+    },
+  },
+)
+```
+
```

Comment:
Maybe add an example using the driver into react with connect (or a link to en example in the connect documentation)